unified-runtime/scripts/core/EXP-ASYNC-ALLOC.rst

```diff
@@ -0,0 +1,105 @@
+<%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+
+.. _experimental-async-allocations:
+
+================================================================================
+Async Allocation Functions
+================================================================================
+
+.. warning::
+
+    Experimental features:
+
+    *   May be replaced, updated, or removed at any time.
+    *   Do not require maintaining API/ABI stability of their own additions over
+        time.
+    *   Do not require conformance testing of their own additions.
+
+
+Motivation
+--------------------------------------------------------------------------------
+
+Asynchronous allocations can allow queues to allocate and free memory between 
+UR command enqueues without forcing synchronization points in the asynchronous 
+command DAG associated with a queue. Through the enqueue-ordering semantics,
+memory allocated within a pool can be reused so as to avoid expensive and 
+redundant calls into the OS, which can improve performance.
+
+API
+--------------------------------------------------------------------------------
+
+Enums
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* ${x}_device_info_t
+    * ${X}_DEVICE_INFO_ASYNC_USM_ALLOCATIONS_EXP
+* ${x}_usm_pool_flags_t
+    * ${X}_USM_POOL_FLAG_USE_NATIVE_MEMORY_POOL_EXP
```

Comment:
I don't get the idea behind this `UR_USM_POOL_FLAG_USE_NATIVE_MEMORY_POOL_EXP` flag. Was it meant to fulfill the async API requirement `Create a memory pool from a user provided USM allocation`? If so, there's currently no API that allows user to provide such pointer for pool creation.

Btw, my comments aren’t meant to delay merging this PR. This discussion will help with the future spec changes in separate PRs.